samples/rust/rust_smp_cpu.rs

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Rust SMP CPU manipulation sample.
+
+use kernel::{cpu::Cpu, error::code::*, prelude::*};
+
+module! {
+    type: RustCpu,
+    name: b"rust_smp_cpu",
+    author: b"Rust for Linux Contributors",
+    description: b"Rust SMP CPU sample",
+    license: b"GPL",
+}
+
+struct RustCpu;
+
+impl kernel::Module for RustCpu {
+    fn init(_name: &'static CStr, _module: &'static ThisModule) -> Result<Self> {
+        pr_info!("Rust SMP CPU sample init.\n");
+
+        let guard = Cpu::lock_current();
+        let id = guard.id();
+        pr_info!("Running on CPU #{id}\n");
+
+        let num_possible = Cpu::num_possible();
+        pr_info!("Max possible CPUs: {num_possible}\n");
+
+        if num_possible == 1 {
+            return Err(ENOSPC);
+        }
+
+        guard.call(|| {
+            pr_info!("Running closure on current locked processor #{id}\n");
+        })?;
+
+        // Assume CPUs zero and one are online.
+        let remote = if id == 0 {
+            Cpu::from_id(1)?
+        } else {
+            Cpu::from_id(0)?
+        };
```

Comment:
Debatable how readable it is, but we could use the `if` as an expression:

```suggestion
        let remote = Cpu::from_id(if id == 0 { 1 } else { 0 })?;
```